tools/storage/segment.py

```diff
@@ -0,0 +1,303 @@
+    def from_bytes(b):
+        """ Deocdes record from bytes """
+        size, v = read_varlong(b, 0)
+        record_attr = b[v]
+        v += 1
+
+        timestamp_delta, v = read_varlong(b, v)
+        offset_delta, v = read_varlong(b, v)
+        key_length, v = read_varlong(b, v)
+
+        key = None
+        if key_length > 0:
+            key = b[v: v + key_length]
+            v += key_length
+
+        value_length, v = read_varlong(b, v)
+        value = b[v:v + value_length]
+        v += value_length
+
+        header_count, v = read_varlong(b, v)
+        headers = {}
+        for _ in range(0, header_count):
+            hkey = None
+            hvalue = None
+            hkey_length, v = read_varlong(b, v)
+            if hkey_length > 0: 
+                hkey = b[v:v + hkey_length]
+                v += hkey_length
+            hvalue_length, v = read_varlong(b, v)
+            if hvalue_length > 0:
+                hvalue = b[v:v + hvalue_length]
+                v += hvalue_length
+            if hkey is not None: 
+                headers[hkey] = hvalue
+        return Record(size, record_attr, timestamp_delta,
+                      offset_delta, key, headers, value)
+
+    @staticmethod
+    def records_from_bytes(bytes, count):
+        records = []
+        for _ in range(0, count):
+            record = Record.from_bytes(bytes)
+            size = record.size + 1
+            bytes = bytes[size:]
+            records.append(record)
+        return records
+
+
+class Batch:
+    """ Batch of Records in a Segment file """
+    def __init__(self, index, header, records):
+        self.index = index
+        self.header = header
+        self.records = records
+        self.__verify_crc()
+
+    def __verify_crc(self):
+        header_crc_bytes = struct.pack(
+            "<" + HDR_FMT_RP_PREFIX_NO_CRC + HDR_FMT_CRC, *self.header[1:])
+        header_crc = crc32c.crc32c(header_crc_bytes)
+        if self.header.header_crc != header_crc:
+            raise CorruptBatchError(self)
+        crc = crc32c.crc32c(self._crc_header_be_bytes())
+        crc = crc32c.crc32c(self.records, crc)
+        if self.header.crc != crc:
+            raise CorruptBatchError(self)
+
+    def last_offset(self):
+        return self.header.base_offset + self.header.record_count - 1
+
+    def _crc_header_be_bytes(self):
+        # encode header back to big-endian for crc calculation
+        return struct.pack(">" + HDR_FMT_CRC, *self.header[5:])
+
+    def parse_records(self):
+        bytes = self.records
+        if self.header.type != 1:
+            return []
+        try:
+            if self.header.attrs == 0:
+                pass # uncompressed
+            elif self.header.attrs == 1:
+                logger.debug("Decompressing gzip batch")
+                bytes = gzip.decompress(self.records)
+            elif self.header.attrs == 2:
+                raise NotImplementedError("Unsupported compression type: Snappy")
```

Comment:
This probably needs implementing before shipping the tool.  Otherwise any client can break the decode by sending in a snappy-encoded batch to the topic (I think?  Unless there's some restriction elsewhere on what encodings are permitted)